src/Core/System/Capabilities.js

```diff
@@ -17,16 +34,33 @@ export default {
         const gl = renderer.context;
         maxTexturesUnits = gl.getParameter(gl.MAX_TEXTURE_IMAGE_UNITS);
 
-        const debugInfo = gl.getExtension('WEBGL_debug_renderer_info');
-        if (debugInfo !== null) {
-            const vendor = gl.getParameter(debugInfo.UNMASKED_VENDOR_WEBGL);
-            if (vendor.indexOf('mesa') > -1 || vendor.indexOf('Mesa') > -1) {
-                maxTexturesUnits = Math.min(16, maxTexturesUnits);
+        const program = gl.createProgram();
+        const glVertexShader = _WebGLShader(renderer, gl.VERTEX_SHADER, SampleTestVS);
+
+        let fragmentShader = `#define SAMPLE ${maxTexturesUnits}\n`;
+        fragmentShader += SampleTestFS;
+
+        const glFragmentShader = _WebGLShader(renderer, gl.FRAGMENT_SHADER, fragmentShader);
+
+        gl.attachShader(program, glVertexShader);
+        gl.attachShader(program, glFragmentShader);
+        gl.linkProgram(program);
+
+        if (gl.getProgramParameter(program, gl.LINK_STATUS) === false) {
+            // eslint-disable-next-line no-console
+            console.warn(`Sampler limit exceeded: it's down from ${maxTexturesUnits} to 16`);
+            if (isFirefox()) {
+                // eslint-disable-next-line no-console
+                console.warn('it can come from a bug mesa/firefox \n\n' +
```

Comment:
You could do a multiline string like this:
```js
console.warn(`It can come from a bug mesa/firefox.
Error compile shader when using more than 16 sampler uniforms,
see https://bugzilla.mozilla.org/show_bug.cgi?id=777028`);
```

Reply:
done